iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/transformation/dag/input/ConstantInputReader.java

```diff
@@ -82,62 +100,27 @@ public boolean isConstantPointReader() {
   }
 
   @Override
-  public YieldableState yield() {
-    return YieldableState.YIELDABLE;
-  }
-
-  @Override
-  public boolean next() {
-    return true;
-  }
-
-  @Override
-  public void readyForNext() {
+  public void consumed(int consumed) {
     // Do nothing
   }
 
   @Override
-  public TSDataType getDataType() {
-    return expression.getDataType();
-  }
-
-  @Override
-  public long currentTime() throws IOException {
-    throw new UnsupportedOperationException();
-  }
-
-  @Override
-  public int currentInt() throws IOException {
-    return cachedInt;
-  }
-
-  @Override
-  public long currentLong() throws IOException {
-    return cachedLong;
-  }
-
-  @Override
-  public float currentFloat() throws IOException {
-    return cachedFloat;
-  }
-
-  @Override
-  public double currentDouble() throws IOException {
-    return cachedDouble;
+  public void consumedAll() {
+    // Do nothing
   }
 
   @Override
-  public boolean currentBoolean() throws IOException {
-    return cachedBoolean;
+  public Column[] current() throws IOException {
+    return new Column[] {cachedColumn};
```

Comment:
Perhaps making cachedColumn Column[] instead of Column could avoid new array each time.

Reply:
Good idea.